lib/libzfs/libzfs_mount.c

```diff
@@ -1595,9 +1596,11 @@ zpool_disable_datasets(zpool_handle_t *zhp, boolean_t force)
 			goto out;
 	}
 
-	for (i = 0; i < used; i++) {
-		if (datasets[i])
-			remove_mountpoint(datasets[i]);
+	if (!hdl->libzfs_force_export) {
+		for (i = 0; i < used; i++) {
+			if (datasets[i])
+				remove_mountpoint(datasets[i]);
+		}
```

Comment:
It's because `rmdir(2)` will attempt I/O for most mountpoints on a given pool being forcibly exported.  Only the top-level (in the sense of the filesystem hierarchy) datasets aren't subject to this constraint. At the same time, I question whether it's worth the effort & code to determine which ones are top-level.

Reply:
Yeah, it's a bit awkward but I think we want to preserve the current behavior if possible.  How about just adding the insight about `rmdir(2)` attempting IO for the mountpoint to the comment since that wasn't obvious to me at least when reading through this code.